internal/controller/mcpserver_controller.go

```diff
@@ -86,6 +97,19 @@ func (r *MCPServerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return ctrl.Result{}, err
 	}
 
+	meta.SetStatusCondition(&mcpServer.Status.Conditions, r.getDeploymentCondition(ctx, r.Client, mcpServer))
+	meta.SetStatusCondition(&mcpServer.Status.Conditions, r.getServiceCondition(ctx, r.Client, mcpServer))
+	meta.SetStatusCondition(&mcpServer.Status.Conditions, r.getRouteCondition(ctx, r.Client, mcpServer))
+
+	overallReady := r.getOverallCondition(mcpServer)
+	meta.SetStatusCondition(&mcpServer.Status.Conditions, overallReady)
+
+	if overallReady.Status != metav1.ConditionTrue {
+		logger.Info("MCPServer not yet fully ready, re-queuing...", "reason", overallReady.Reason, "message", overallReady.Message)
+		return ctrl.Result{RequeueAfter: 15 * time.Second}, nil
+	}
```

Comment:
Rather than using `defer` above, I would set the status here so you can handle the error in the way the controller expects.
```suggestion
	overallReady := r.getOverallCondition(mcpServer)
	meta.SetStatusCondition(&mcpServer.Status.Conditions, overallReady)
	
	// Only update the status if it changed
	if !reflect.DeepEqual(originalStatus, &mcpServer.Status) {
		logger.Info("Status has changed, attempting to update")
		if err = r.Status().Update(ctx, mcpServer); err != nil {
			logger.Error(err, "unable to update MCPServer status")
			return ctrl.Result{}, err
		}
		logger.Info("Successfully updated MCPServer status")
	}

	if overallReady.Status != metav1.ConditionTrue {
		logger.Info("MCPServer not yet fully ready, re-queuing...", "reason", overallReady.Reason, "message", overallReady.Message)
		return ctrl.Result{RequeueAfter: 15 * time.Second}, nil
	}
```